mitodl/mitxonline
# Added unenroll button to dashboard

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #217

#### What's this PR do?
- Adds an unenroll button to the dashboard so users can unenroll themselves from courses

#### How should this be manually tested?
- Make sure the nav menu looks correct in all screen widths, with the dropdown menu open or hidden
- Go to the dashboard and unenroll from a course run using the dot menu. You should see a message that you successfully unenrolled, your enrollment record should be set to `active=False` and `change_status="unenrolled"`, and if your edX instance has the equivalent course run, you should see a successful unenrollment request to edX in your logs
- Return a 4**/5** response from the `destroy` method of the user enrollment viewset, then try unenrolling from a different course run. You should see an error message.

#### Any background context you want to provide?
- This PR involves a pretty heavy CSS refactor. My initial goal was just to pull out some common styles for the dropdown menu, which was already implemented for the logged-in user menu. In the process of doing that, I discovered a large amount of unnecessary or misplaced styling code (the styles for the user menu were spread across 3 files in a way that didn't really make sense)

#### Screenshots (if appropriate)
![ss 2021-10-15 at 14 48 57 ](https://user-images.githubusercontent.com/14932219/137545543-5bf59df7-2b25-4913-a3ec-0db8488b6854.png)
![ss 2021-10-15 at 14 48 01 ](https://user-images.githubusercontent.com/14932219/137545549-6ae58d86-039a-4eed-abca-bc9f45bba683.png)
![ss 2021-10-15 at 14 51 24 ](https://user-images.githubusercontent.com/14932219/137545734-97af7179-a620-47a4-9e3b-b3d3249bf4ab.png)

**NOTE:** The dashboard cards up until now have only shown the image-on-top-of-title UI in browser widths under 540px. Since this dot menu is now here occupying space on the title line, I changed that so the same UI is shown up to 720px (aka "sm" screen width).

<img width="763" alt="ss 2021-10-26 at 15 49 09 " src="https://user-images.githubusercontent.com/14932219/139102799-25c0c256-8db5-48c1-beeb-15dccc2d5399.png">
